### PR TITLE
Update boost test lecture

### PR DESCRIPTION
**Pull request by uekerman**
## Description

- I updated the lecture on Boost.Test and added a small demo section on tests in preCICE.
- I did not yet update the exercise. This might come in a separate PR tomorrow only.

## Checklist

- [x] I made sure that the markdown files are formatted properly.
- [ ] I made sure that all hyperlinks are working.

- [x] I added the new content to the `README.md` inside the topics subdirectory, e.g., `01_version_control/README.md`.
- [x] I added the new content to the `timetable.md` in the root of this repository.

### CODE REVIEW

File: 05_testing_and_ci/boost_testing_precice_demo.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+- Look at `math/tests/DifferenceTest.cpp`
+    - Imports `testing/Testing.hpp`, there we handle UTF imports
+    - Test suite for `math` namespace and test suite per file (here `math/differences.hpp`)
+    - `BOOST_CHECK` actually not recommended to use. Is used internally by `BOOST_TEST`.
+    - Powerful macro `PRECICE_TEST(1_rank)`; here it means that this test is run on one MPI rank
+        - Tests are normally run on 4 MPI ranks to test parallel implementation and to mimic different coupled solvers
+
+## Unit vs. Integration Tests
+
+- Clear separation in preCICE: integration tests only directly use API of preCICE.
```
Comment (IshaanDesai) [inline 05_testing_and_ci/boost_testing_precice_demo.md:18]:
```suggestion
- Clear separation in preCICE: only integration tests directly use API of preCICE.
```
The suggestion is only valid if you meant to say that integration tests directly use API of preCICE, and unit tests do not.

File: 05_testing_and_ci/boost_testing_sidemade_demo.md [hunk 2]
```diff
@@ -95,18 +96,18 @@ Repository: [testing boost exercise – demo-start branch](https://github.com/Si
     BOOST_TEST(x == expectedX);
     ```
 
-- adjust `CMakeLists.txt`
-    - add `"${SRC_FILES}"` to test executable
-    - makes headers visible `target_include_directories(testsidemade PRIVATE ${PROJECT_SOURCE_DIR}/src)`
+- Adjust `CMakeLists.txt`
```
Comment (IshaanDesai) [inline 05_testing_and_ci/boost_testing_sidemade_demo.md:99]:
```suggestion
- Adjust `CMakeLists.txt`:
```
to be consistent with what is done above.

File: 05_testing_and_ci/boost_testing_sidemade_demo.md [hunk 3]
```diff
@@ -148,28 +158,35 @@ Repository: [testing boost exercise – demo-start branch](https://github.com/Si
...
 
-- For floating point values, we need a tolerance.
-    - use decorator `*boost::unit_test::tolerance(1e-12)` in suite
+- Seems that for floating point values, we need a tolerance. Use decorator, directly add to suite:
+
+    ```cpp
+    BOOST_FIXTURE_TEST_SUITE(MatrixSolverTests, MatrixSolverFixture, *boost::unit_test::tolerance(1e-12))
+    ```
+
+- Better change all comparisons to explicit calls to entries. Or overwrite Eigen implementation.
```
Comment (IshaanDesai) [inline 05_testing_and_ci/boost_testing_sidemade_demo.md:181]:
```suggestion
- Better change all comparisons to explicit calls into entries. Or overwrite Eigen implementation.
```
This suggestion is rather a matter of style, I find two "to"s weird.

File: 05_testing_and_ci/boost_testing_sidemade_demo.md [hunk 4]
```diff
@@ -188,16 +205,16 @@ Repository: [testing boost exercise – demo-start branch](https://github.com/Si
     BOOST_AUTO_TEST_SUITE_END()
     ```
 
-- `cp ../data/config.yml testconfig.yml` and explain what test should do
-- expected data:
+- `cp ../data/config.yml testconfig.yml` and explain what test should do TODO
```
Comment (IshaanDesai) [inline 05_testing_and_ci/boost_testing_sidemade_demo.md:208]: Leftover TODO

**Review by IshaanDesai — APPROVED** [review]
I went over all the changes, just some minor comments from my side.

File: 05_testing_and_ci/boost_testing_precice_demo.md (hunk 1 above)
Comment (uekerman) [reply 05_testing_and_ci/boost_testing_precice_demo.md:18]: No, I meant indeed that integration tests use nothing but the API of preCICE. (no lower-level code)

**Pull request merged**